war/src/main/java/io/lumeer/engine/controller/SecurityFacade.java

```diff
@@ -51,9 +52,296 @@
+      String userOrGroupKey = userOrGroupKey(user);
+      String userOrGroupName = userOrGroupName(user, group);
+
+      Map<String, Object> filter = collectionFilter(projectId, collectionName);
+
+      dataStorage.removeItemFromArray(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(filter),
+            dataStorageDialect.concatFields(
+                  LumeerConst.Security.ROLES_KEY,
+                  role,
+                  userOrGroupKey),
+            userOrGroupName);
+   }
+
+   public void removeViewUserRole(String projectId, int viewId, String user, String role) {
+      removeViewRole(projectId, viewId, user, null, role);
+   }
+
+   public void removeViewGroupRole(String projectId, int viewId, String group, String role) {
+      removeViewRole(projectId, viewId, null, group, role);
+   }
+
+   private void removeViewRole(String projectId, int viewId, String user, String group, String role) {
+      String userOrGroupKey = userOrGroupKey(user);
+      String userOrGroupName = userOrGroupName(user, group);
+
+      Map<String, Object> filter = viewFilter(projectId, viewId);
+
+      dataStorage.removeItemFromArray(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(filter),
+            dataStorageDialect.concatFields(
+                  LumeerConst.Security.ROLES_KEY,
+                  role,
+                  userOrGroupKey),
+            userOrGroupName);
+   }
+
+   private Map<String, Object> projectFilter(String projectId) {
+      Map<String, Object> filter = new HashMap<>();
+      filter.put(LumeerConst.Security.PROJECT_ID_KEY, projectId);
+      filter.put(LumeerConst.Security.TYPE_KEY, LumeerConst.Security.TYPE_PROJECT);
+      return filter;
+   }
+
+   private Map<String, Object> collectionFilter(String projectId, String collectionName) {
+      Map<String, Object> filter = new HashMap<>();
+      filter.put(LumeerConst.Security.PROJECT_ID_KEY, projectId);
+      filter.put(LumeerConst.Security.TYPE_KEY, LumeerConst.Security.TYPE_COLLECTION);
+      filter.put(LumeerConst.Security.COLLECTION_NAME_KEY, collectionName);
+      return filter;
+   }
+
+   private Map<String, Object> viewFilter(String projectId, int viewId) {
+      Map<String, Object> filter = new HashMap<>();
+      filter.put(LumeerConst.Security.PROJECT_ID_KEY, projectId);
+      filter.put(LumeerConst.Security.TYPE_KEY, LumeerConst.Security.TYPE_VIEW);
+      filter.put(LumeerConst.Security.VIEW_ID_KEY, viewId);
+      return filter;
+   }
+
+   private String userOrGroupKey(String user) {
+      return user != null ? LumeerConst.Security.USERS_KEY : LumeerConst.Security.GROUP_KEY;
+   }
+
+   private String userOrGroupName(String user, String group) {
+      return user != null ? user : group;
+   }
+
+   /********* OLD **********/
```

Comment:
I haven't removed it yet because then I would have to change many other facades which use those methods. I will of course do it once all new methods are finished.